src/tests/algorithms/test_generic.py

```diff
@@ -40,3 +40,169 @@ def test_rescale_image(self):
+    ]
+
+    _known_inputs_outputs_2D = [
+        ([[],
+          []],
+         [[],
+          []]),
+
+        ([[5.],
+          [5.]],
+         [[5.],
+          [5.]]),
+
+        (np.array([[4., 6.],
+                   [4., 6.]]),
+         np.array([[14., 16.],
+                   [14., 16.]]) / 3.),
+
+        (np.ones((5, 5)),
+         np.ones((5, 5))),
+
+        (np.array([[1., 2., 1.],
+                   [1., 2., 1.]]),
+         np.array([[4., 4., 4.],
+                   [4., 4., 4.]]) / 3.),
+
+        (np.array([[0., 1., 2., 3.],
+                   [0., 1., 2., 3.]]),
+         np.array([[1., 3., 6., 8.],
+                   [1., 3., 6., 8.]]) / 3.),
+
+        (np.array([[-1,  1., 2., 3., -3.],
+                   [-1,  1., 2., 3., -3.]]),
+         np.array([[-1., 2., 6., 2., -3.],
+                   [-1., 2., 6., 2., -3.]]) / 3.),
+
+        (np.array([[-1,   1., -1., 1., -1., 1.],
+                  [-1,   1., -1., 1., -1., 1.]]),
+         np.array([[-1., -1.,  1, -1.,  1., 1.],
+                   [-1., -1.,  1, -1.,  1., 1.]]) / 3.),
+
+        (np.array([[-0.5, 1.5, 2.5, 3.5, -2.5],
+                   [-0.5, 1.5, 2.5, 3.5, -2.5]]),
+         np.array([[ 0.5, 3.5, 7.5, 3.5, -1.5],
+                   [ 0.5, 3.5, 7.5, 3.5, -1.5]]) / 3.),
+    ]
+
+
+    def assertFilterOutput(self, signal, kernel, out_exp):
+        """Assert that signal generates output close to expectation"""
+        with self.subTest(signal=signal):
+            out_fnd = boxcar_filter(signal, kernel)
+            msg = ''
+            try:
+                np.testing.assert_allclose(out_fnd, out_exp)
+            except AssertionError as e:
+                msg = '\nsignal: %s\ngives wrong filtered output:' % (signal,)
+                msg += str(e).replace('x:', '  actual:').replace('y:', 'expected:')
+
+            if msg:
+                self.fail(msg)
+
+
+    def test_float_inputs(self):
+        kernel_1D = TestBoxcarFilter._kernel_size_1D
+        for signal, out_exp in TestBoxcarFilter._known_inputs_outputs_1D:
+            self.assertFilterOutput(signal, kernel_1D, out_exp)
+
+        kernel_2D = TestBoxcarFilter._kernel_size_2D
+        for signal, out_exp in TestBoxcarFilter._known_inputs_outputs_2D:
+            self.assertFilterOutput(signal, kernel_2D, out_exp)
+
+    def test_int_inputs(self):
+        kernel_1D = TestBoxcarFilter._kernel_size_1D
+        for signal, out_exp in TestBoxcarFilter._known_inputs_outputs_1D:
+            int_signal = np.array(signal, dtype=int)
+            try:
+                np.testing.assert_allclose(int_signal, signal)
+            except AssertionError:
+                continue
+
+            self.assertFilterOutput(int_signal, kernel_1D, out_exp)
+
+        kernel_2D = TestBoxcarFilter._kernel_size_2D
+        for signal, out_exp in TestBoxcarFilter._known_inputs_outputs_2D:
+            int_signal = np.array(signal, dtype=int)
+            try:
```

Comment:
Replace the `try-except` block with
```
with self.assertRaises(AssertionError):
   np.testing.assert_allclose(int_signal, signal)
```
(similar for some other methods)

Reply:
The goal of the code is to only process those arrays whose float values are actually integers. The suggested replacement does not replicate that behavior. I have replaced the `np.testing.assert_allclose() + try/except` 
with 
`if np.allclose():`